src/lib.rs

```diff
@@ -374,94 +375,100 @@ fn pick_mesh(
     let cursor_pos_ndc: Vec3 =
         ((cursor_pos_screen / screen_size) * 2.0 - Vec2::from([1.0, 1.0])).extend(1.0);
 
-    // Get the view transform and projection matrix from the camera
-    let mut camera_matrix = Mat4::zero();
-    let mut projection_matrix = Mat4::zero();
-    for (transform, camera) in &mut camera_query.iter() {
-        camera_matrix = *transform.value();
-        projection_matrix = camera.projection_matrix;
-    }
-    let (_, _, camera_position) = camera_matrix.to_scale_rotation_translation();
+    // collect and calculate pick_ray and camera_position from all cameras
+    let mut rays: HashMap<Entity, (Ray3D, Vec3)> = HashMap::new();
```

Comment:
The camera position Vec3 here is redundant - the ray origin is the camera position. You can simplify this to a Hashmap of type <Entity, Ray3D>